tiny_cnn/core/tiny_backend.h

```diff
@@ -54,13 +54,13 @@ class tiny_backend : public math_backend {
       vec_t&       a   = *out_data[1];
       const vec_t &in  = *((*conv_layer_worker_storage_)[index].prev_out_padded_); // input // NOLINT
 
-      cnn_size_t idx = 0;
       std::fill(a.begin(), a.end(), float_t(0));
 
       for_i(layer_->get_parallelize(), params_->out.depth_, [&](int o) {
           for (cnn_size_t inc = 0; inc < params_->in.depth_; inc++) {
               if (!params_->tbl.is_connected(o, inc)) continue;
 
+              cnn_size_t idx = 0;
               idx = params_->in.depth_ * o + inc;
```

Comment:
@nyanp Putting the idx var inside the parallel for should fix the data race issue, right?